### ISSUE TITLE
Show color of Sass variables?

### ISSUE THREAD

**Issue by LoDef** [sergiirocks/vscode-ext-color-highlight#29]
Would it be possible to show the color of a Sass variable?
For example, if I have a variable declared like this in my Sass setup:

`$primaryColor: #77baa8;`

which is then used like this:

`&.div { background-color: $primaryColor; }`

It would be very handy if `$primaryColor` in the latter example got color highlighted with #77baa8

**Comment by MaciejJanyska** [comment]
👍  +1

**Comment by Jordan4jc** [comment]
I see in `v2.3.0` the changelog states this works and you can specify a Configuration option for the sass imports lookup folders? How does this lookup config work? Is there an example of it somewhere?

**Comment by LoDef** [comment]
@Jordan4jc 
From the Settings-file:
```
// Array of absolute paths to search while perform file lookups.
  "color-highlight.sass.includePaths": [],
```
No example found though, so I tried various values, both relative and absolute, with front- or backslash, even rebooted VSCode, but no success. It only seems to work when the definition is in the same scss-file, it can't find the definition outside the current file.
```
    "color-highlight.sass.includePaths": [
      "C:/xampp/htdocs/mysite/theme/scss",
      "src/styles", 
      "scss", 
      "/scss/utils"]
```

**Comment by goodguyry** [comment]
I'm looking through the code for help understanding what `includePaths` is expecting and I don't see any references to it...

**Comment by stormwarning** [comment]
The `Window` Output panel was showing this error:
```
[renderer1] [error] ENOTDIR: not a directory, lstat '/Users/me/project/styles/_vars.color.scss/color.scss'
```

I was able to get it working with external variables by `@import`ing the filename explicitly. For example:
```diff
- @import 'vars.color';

+ @import '_vars.color.scss';
```

Not ideal, and in my case, flags linter errors. I tried looking into a fix but I'm not sure how this project is even set up — I have version 2.3.0 installed apparently, but GitHub only shows version 2.2.0... 🤔

**Comment by mattaningram** [comment]
Why does this require configuring include paths anyway? The Atom equivalent of this plugin just seems to grab variables from any SCSS file in your project.

**Comment by Mei152** [comment]
Agree with @mattaningram.  I am trying to migrate to vscode from Atom.  This is a feature I really liked in Atom.

**Comment by nizzu1102** [comment]
Color Highlight   -   https://marketplace.visualstudio.com/items?itemName=naumovs.color-highlight  --- it helps me do that.

**Comment by nizzu1102** [comment]
Color highlight, does that function, but in my case in each file I have to refer to the file where the color variables are with a comment because if not the sinter linter gives me an error,
I import it by calling the file with the color variables (_color-var.scss).
thus.
// @import '../color-var.color';

Color highlight hace esa función, pero en mi caso en cada archivo tengo que hacer referencia al archivo donde se encuentran las variables de color con un comentario porque sino el linter de sass me da error,
la importación la hago  llamando al archivo con las variables de color ( _color-var.scss )
de esta forma
// @import '../color-var.color';

**Comment by LiamWinterton** [comment]
Is there any documentation for includePaths yet, as this still does not work.

**Comment by sethstha** [comment]
Why don't you guys try Colorize?

**Comment by qbconcept** [comment]
> Why don't you guys try Colorize?

I would need something more subtle than colorize (or I can't see the setting to make it more subtle). I can't remember if what I have is vanilla VS Code, but this small square preceding HEX color is perfect. It would be nice to have the same with Sass variables.

**Comment by ronilaukkarinen** [comment]
@qbconcept https://github.com/KamiKillertO/vscode-colorize/issues/274

**Comment by silviajoy** [comment]
I agree this feature would be very helpful to use without the need to explicitly import the variable file

**Comment by panyann** [comment]
> Why don't you guys try Colorize?

Because it doesn't support colorizing the sass variables?

**Comment by EzePerucca** [comment]
Yes it does.

![image](https://user-images.githubusercontent.com/39660083/157513082-b51aa1fa-dbf9-4644-abc8-1780330a9a1f.png)

Open the settings.json and add those lines
![image](https://user-images.githubusercontent.com/39660083/157513013-c18ecd5f-785a-43ee-9389-6f29d7ec9308.png)

**Comment by albertochigua** [comment]
**For Sass...**
_Use extension..._
![Screenshot from 2022-04-21 23-11-13](https://user-images.githubusercontent.com/20590139/164607483-1034bdf5-ac92-448d-b02c-7430dd245f8b.png)

_In settings..._
![Screenshot from 2022-04-21 23-11-24](https://user-images.githubusercontent.com/20590139/164607544-b050d774-90b5-49e7-b9d5-f04bb3872051.png)

_add "SASS"..._
![Screenshot from 2022-04-21 23-10-44](https://user-images.githubusercontent.com/20590139/164607640-bc4091fa-c397-4aae-b911-748ec57cf672.png)

result...
![Screenshot from 2022-04-21 23-14-02](https://user-images.githubusercontent.com/20590139/164607694-6e1fadad-7a3d-4a37-89b6-522c321e6757.png)

**Comment by Serengia** [comment]
> **For Sass...** _Use extension..._ ![Screenshot from 2022-04-21 23-11-13](https://user-images.githubusercontent.com/20590139/164607483-1034bdf5-ac92-448d-b02c-7430dd245f8b.png)
> 
> _In settings..._ ![Screenshot from 2022-04-21 23-11-24](https://user-images.githubusercontent.com/20590139/164607544-b050d774-90b5-49e7-b9d5-f04bb3872051.png)
> 
> _add "SASS"..._ ![Screenshot from 2022-04-21 23-10-44](https://user-images.githubusercontent.com/20590139/164607640-bc4091fa-c397-4aae-b911-748ec57cf672.png)
> 
> result... ![Screenshot from 2022-04-21 23-14-02](https://user-images.githubusercontent.com/20590139/164607694-6e1fadad-7a3d-4a37-89b6-522c321e6757.png)

Thank you so much.

**Comment by Serengia** [comment]
Thank you so much @albertochigua

**Comment by Azragh** [comment]
@albertochigua This sadly only works for directly assigned variables. As soon as a variable gets assigned to another one (or if you try to mix colors or similar) it doesn't work and is therefore useless to me. In Atom (with the "pigments" package) I can do this:

![image](https://user-images.githubusercontent.com/15014142/217789261-7cc5e1f6-a192-4d01-b2cd-2f03c8e3a078.png)

» But the Atom Project was discontinued in dec 2022 so I'm looking forward to use VSC.

I'm desperately searching for an extension which is able to follow color functions and stuff.. Is there anything comparable?

**Comment by Azragh** [comment]
Any updates on this? I am desperately searching, got so far that I stumbled across my own comment here..

So my VS Code just looks like this when using Colorize and the suggested settings:

![image](https://user-images.githubusercontent.com/15014142/227233710-d21d0aa4-1a88-438c-a5a5-9b09eaf5a6e3.png)

I also tried Color Highlight and many others, tinkered with the settings, asked questions on stackoverflow and so on, but no success.

**Comment by serengia** [comment]
> only works for directly assigned variables. As soon as a variable gets assigned to another one

Hi @Azragh , while I don't fully understand what you exactly mean by the above statement... I have been using `colorize` Vs extension for over 6 months now on a project with multiple pages. After installing the extension, ensure you add SASS in your json settings as  @albertochigua highlighted above.

**Comment by Azragh** [comment]
Hi @serengia. Funny, I stumbled across my own comment again. Sorry for not responding. What I meant with "only directly assigned variables" should be clear if you compare the two screenshots from my comments. Here you can see how Atom always displays the correct color value even with multiple instantiation / recalculation of variables:

![image](https://github.com/user-attachments/assets/3cae5d9b-cd47-4f54-ba51-3350f193a8b2)

I just can't get Colorize to do more than this:

![image](https://github.com/user-attachments/assets/4ae0d7cc-2ace-4cb3-a7af-3bc6f77afc42)
